xdsl/dialects/memref.py

```diff
@@ -528,14 +528,12 @@ class ExtractStridedMetaDataOp(IRDLOperation):
 
     irdl_options = [AttrSizedResultSegments()]
 
-    @staticmethod
-    def from_source_memref(
-        source: SSAValue | Operation, source_type: MemRefType[Attribute]
-    ):
+    def __init__(self, source: SSAValue | Operation):
         """
         Create an ExtractStridedMetaDataOp that extracts the metadata from the
         operation (source) that produces a memref.
         """
+        source_type = SSAValue.get(source).type
```

Comment:
You can slap an assert here `assert isa(source_type, MemRefType[Attribute])` or something like that